nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToCSV.java

```diff
@@ -0,0 +1,335 @@
+            "Existing flowfile context will be overwritten. 'CSVData' will not be written to at all (neither null nor empty string).");
+    static final AllowableValue OUTPUT_NEW_ATTRIBUTE= new AllowableValue("flowfile-attribute", "flowfile-attribute", "The resulting CSV string will be placed into a new flowfile" +
+            " attribute named 'CSVData'.  The content of the flowfile will not be changed.");
+
+    public static final PropertyDescriptor ATTRIBUTES_LIST = new PropertyDescriptor.Builder()
+            .name("attribute-list")
+            .displayName("Attribute List")
+            .description("Comma separated list of attributes to be included in the resulting CSV. If this value " +
+                    "is left empty then all existing Attributes will be included. This list of attributes is " +
+                    "case sensitive and supports attribute names that contain commas. If an attribute specified in the list is not found it will be emitted " +
+                    "to the resulting CSV with an empty string or null depending on the 'Null Value' property. " +
+                    "If a core attribute is specified in this list " +
+                    "and the 'Include Core Attributes' property is false, the core attribute will be included. The attribute list " +
+                    "ALWAYS wins.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor ATTRIBUTES_REGEX = new PropertyDescriptor.Builder()
+            .name("attributes-regex")
+            .displayName("Attributes Regular Expression")
+            .description("Regular expression that will be evaluated against the flow file attributes to select "
+                    + "the matching attributes. This property can be used in combination with the attributes "
+                    + "list property.  The final output will contain a combination of matches found in the ATTRIBUTE_LIST and ATTRIBUTE_REGEX.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.createRegexValidator(0, Integer.MAX_VALUE, true))
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor DESTINATION = new PropertyDescriptor.Builder()
+            .name("destination")
+            .displayName("Destination")
+            .description("Control if CSV value is written as a new flowfile attribute 'CSVData' " +
+                    "or written in the flowfile content.")
+            .required(true)
+            .allowableValues(OUTPUT_NEW_ATTRIBUTE, OUTPUT_OVERWRITE_CONTENT)
+            .defaultValue(OUTPUT_NEW_ATTRIBUTE.getDisplayName())
```

Comment:
The "write to attribute" is a cool concept, I thought it might also be beneficial in other areas (including UpdateAttribute), so I wrote up https://issues.apache.org/jira/browse/NIFI-5276 to cover it (and will soon have a PR). Were it included, would it preclude you from adding such a property to this processor? If not then no worries, just wanted to mention it, in case the other Jira would take care of that use case instead.

Reply:
Nevermind about NIFI-5276, turns out you can already do this with allMatchingAttributes. Since that is the case, do you think it is still useful to have the same type of feature here, or should it just support flowfile content as a destination? The latter would make it a little easier to configure the processor as there wouldn't be a Destination property, and the associated doc would be unnecessary as well

Reply:
we are writting two attributes, the CSVSchema(attribute names) and the CSVData(attribute values).  it looks like allMatchingAttributes will take care of the CSVData attribute nicely, but I don't see any option to handle the schema attribute.  I don't see anything like allMatchingAttributeNames, which would take care of the CSVSchema part.

Reply:
Good point. It would be nice to have something in EL that matches on attribute names and returns them instead of evaluating them to their values, I'll take a look at what that would entail. In the meantime it's cool to have that here, running some final tests but things are looking good :)